### PR TITLE
Move DTH test infrastructure to Microsoft.Dnx.Testing.Framework

### PR DESCRIPTION
**Pull request by troydai**
@JunTaoLuo 

/cc @davidfowl

### CODE REVIEW

File: test/Microsoft.Dnx.Testing.Framework/DnxSdk/Dth/DthMessageTypes.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Dnx.Testing.Framework.DesignTimeHost
+{
+    public class DthMessageTypes
+    {
+        public const string ProtocolVersion = nameof(ProtocolVersion);
```
Comment (troydai) [inline test/Microsoft.Dnx.Testing.Framework/DnxSdk/Dth/DthMessageTypes.cs]: I'll gradually add more test type names to this class.

**Comment by rynowak** [comment]
Should this be called `Microsoft.Dnx.Testing.Extensions`?

**Comment by JunTaoLuo** [comment]
sigh... we just did a rename, but okay we'll do that another time. In terms of this PR, :shipit:

**Comment by rynowak** [comment]
> sigh... we just did a rename, but okay we'll do that another time. In terms of this PR

:trollface: - not a real suggestion

**Pull request merged**